### ISSUE TITLE
cli commands preview & up, fail on gcp stacks predating 3.9.0 that contain cloud run resources using a vpc-access-connector

### ISSUE THREAD

**Issue by CBellNZ** [pulumi/pulumi#14335]
### What happened?

My team heavily uses GCP Cloud Run resources with vpc-access-connectors. After the 3.9.0 release all of our cloud run deployments running on bitbucket have started to fail and we are seeing the same behavior locally. 

After downgrading to 3.89.0 the deployments began to work again.

### Example

Deploy the template below with pulumi cli version `3.89.0`
Upgrade to pulumi `3.9.1`
Run `pulumi preview` or `pulumi up `
Note the following error
```
 gcp:cloudrun:Service (test-app):
    error: expected property name after '.'
```
remove the vpc-access-connector line and run `pulumi preview` again
Note that the error is now gone.

**Sample Template**
```
const cloudrunService = new gcp.cloudrun.Service("test-app", {
  autogenerateRevisionName: true,
  template: {
    spec: {
      containers: [
        {
          image: [IMAGE_NAME],
        }, 
      ],
    },
    metadata: {
      annotations: {
        'run.googleapis.com/vpc-access-connector': 'projects/[PROJECT_ID]/locations/australia-southeast1/connectors/serverless-vpc-connector',
      },
    },
  },
  traffics: [
    {
      latestRevision: true,
      percent: 100,
    },
  ],
```
 

### Output of `pulumi about`

 pulumi about
CLI          
Version      3.90.1
Go Version   go1.21.3
Go Compiler  gc

Plugins
NAME    VERSION
docker  4.4.4
gcp     6.67.0
nodejs  unknown

Host     
OS       ubuntu
Version  20.04
Arch     x86_64

This project is written in nodejs: executable='/home/chris/.nvm/versions/node/v18.16.0/bin/node' version='v18.16.0'

Current Stack: likefamily/issue/dev

TYPE                          URN
pulumi:pulumi:Stack           urn:pulumi:dev::issue::pulumi:pulumi:Stack::issue-dev
pulumi:providers:pulumi       urn:pulumi:dev::issue::pulumi:providers:pulumi::default
pulumi:pulumi:StackReference  urn:pulumi:dev::issue::pulumi:pulumi:StackReference::likefamily/gcp-infrastructure-secrets/development
pulumi:pulumi:StackReference  urn:pulumi:dev::issue::pulumi:pulumi:StackReference::likefamily/gcp-organisation-foundations/development
pulumi:pulumi:StackReference  urn:pulumi:dev::issue::pulumi:pulumi:StackReference::likefamily/gcp-core-infrastructure/development
pulumi:providers:gcp          urn:pulumi:dev::issue::pulumi:providers:gcp::default_6_67_0
pulumi:providers:docker       urn:pulumi:dev::issue::pulumi:providers:docker::default_4_4_4
docker:index/image:Image      urn:pulumi:dev::issue::docker:index/image:Image::
gcp:cloudrun/service:Service  urn:pulumi:dev::issue::gcp:cloudrun/service:Service::test-app

Found no pending operations associated with likefamily/dev

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/chris-likefamily
User           chris-likefamily
Organizations  chris-likefamily, likefamily
Token type     personal

Dependencies:
NAME            VERSION
@pulumi/gcp     6.67.0
@pulumi/pulumi  3.90.0
@pulumi/docker  4.4.4

Pulumi locates its logs in /tmp by default

### Additional context

This error seems to be related to this change
https://github.com/pulumi/pulumi/pull/14257

The error we are seeing seems to be coming from this exception https://github.com/pulumi/pulumi/pull/14257/files#diff-09c7fac151f73951fd1214bb599551d9eb9fe2d614409319fa9e2fa3ca1981b5R64

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Frassle** [comment]
Looks like a bug in the property path parser due to the "." in some of the property names.
Very closely related to https://github.com/pulumi/pulumi/issues/9054 which just got pinged as well.

**Comment by davidedisa** [comment]
Hi

I am having the same trouble deploying a instance in Oracle Cloud with python
If i define multiples defined_tags

defined_tags={"datos_clientes.BBDD": "ENTERPRISE", "datos_clientes.DATACENTER": "DEDICADO"} )

i get this error 

   error: expected property name after '.'

However i am using pulumi 3.91


CLI
Version      3.91.0
Go Version   go1.21.3
Go Compiler  gc

Plugins
NAME    VERSION
oci     1.7.0
python  unknown

Host
OS       ubuntu
Version  22.04
Arch     x86_64

This project is written in python: executable='/usr/bin/python3' version='3.10.12'

david@PTDAVIDGG:~/pulumi$ ./venv/bin/pip show pulumi
Name: pulumi
Version: 3.91.0
Summary: Pulumi's Python SDK
Home-page: https://github.com/pulumi/pulumi
Author:
Author-email:
License: Apache 2.0
Location: /home/david/pulumi/venv/lib/python3.10/site-packages
Requires: dill, grpcio, protobuf, pyyaml, semver, six
Required-by: pulumi-oci


Tnks

**Comment by Frassle** [comment]
@davidedisa can you open a new issue for that and if possible include output from the deployment that failed and if possible debug logs (`-v10 --logtostderr`).
